atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraVerifier.java

```diff
@@ -317,22 +324,31 @@ static void currentRfOnKeyspaceMatchesDesiredRf(CassandraClient client, Cassandr
         sanityCheckReplicationFactor(ks, config, dcs);
     }
 
-    private static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
-        Map<String, String> strategyOptions = Maps.newHashMap(ks.getStrategy_options());
-        for (String dc : dcs) {
-            if (strategyOptions.get(dc) == null) {
+    static void sanityCheckReplicationFactor(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs) {
+        checkRfsSpecified(config, dcs, ks.getStrategy_options());
+        checkRfsMatchConfig(ks, config, dcs, ks.getStrategy_options());
+    }
+
+    private static void checkRfsSpecified(CassandraKeyValueServiceConfig config, Set<String> dcs,
+            Map<String, String> strategyOptions) {
+        for (String dataCenter : dcs) {
+            if (strategyOptions.get(dataCenter) == null) {
                 logErrorOrThrow("The datacenter for this cassandra cluster is invalid. "
-                        + " failed dc: " + dc
-                        + "  strategyOptions: " + strategyOptions, config.ignoreDatacenterConfigurationChecks());
+                        + " failed dc: " + dataCenter + "  strategyOptions: " + strategyOptions,
+                        config.ignoreDatacenterConfigurationChecks());
             }
         }
+    }
 
-        // if currentRF of currentDC != our atlas configured RF) { worry about user skipping RF levels improperly }
-        if (Integer.parseInt(strategyOptions.get(dcs.iterator().next())) != config.replicationFactor()) {
-            throw new UnsupportedOperationException("Your current Cassandra keyspace (" + ks.getName()
-                    + ") has a replication factor not matching your Atlas Cassandra configuration."
-                    + " Change them to match, but be mindful of what steps you'll need to"
-                    + " take to correctly repair or cleanup existing data in your cluster.");
+    private static void checkRfsMatchConfig(KsDef ks, CassandraKeyValueServiceConfig config, Set<String> dcs,
+            Map<String, String> strategyOptions) {
+        for (String dataCenter : dcs) {
```

Comment:
Might be that the old check was correct, if it is guaranteed that the numbers are always consistent, but again this is easier to reason about since it's what you want even without the added condition